cmd/tailscale/cli/update.go

```diff
@@ -326,6 +334,63 @@ func updateDebianAptSourcesListBytes(was []byte, dstTrack string) (newContent []
 	return buf.Bytes(), nil
 }
 
+func (up *updater) updateArch() (err error) {
+	if os.Geteuid() != 0 {
+		return errors.New("must be root; use sudo")
+	}
+
+	defer func() {
+		if err != nil && !errors.Is(err, errUserAborted) {
+			err = fmt.Errorf(`%w; you can try updating using "pacman --sync --refresh tailscale"`, err)
+		}
+	}()
+
+	out, err := exec.Command("pacman", "--sync", "--refresh", "--info", "tailscale").CombinedOutput()
```

Comment:
This is going to break peoples systems as partial updates are not supported on Arch, it's everything or nothing. So this will at some point lead to someone installing incompatible so-name updates on their system and hose it.

You have access to a reasonably responsive Arch maintainer, please reach out before implementing things like this.

Reply:
As this command in short it is the prime example for a partial update `pacman -Syi tailscale`, more info on that here: https://wiki.archlinux.org/title/System_maintenance#Partial_upgrades_are_unsupported

Reply:
Alternatively you don't have to parse the whole information snippet but instead use `--print-format`.

```
pacman -S tailscale --print-format "%v"
```

Reply:
Thanks for your comments!
https://wiki.archlinux.org/title/System_maintenance#Partial_upgrades_are_unsupported talks mostly about libraries. AFAIK the tailscale package itself doesn't contain any shared libraries and only depends on libc.
Is it still risky to update this kind of "leaf" package?

As for the general topic of auto-updates, is there any "standard" way to apply updates in the background for a rolling release like Arch? Other than writing a cronjob, of course.

We would really like to keep all tailscale clients on recent-ish versions, both for general bug/security fixes and for reducing the backwards-compatibility kludges in APIs that tailscale clients talk to.
But we also want to play nice with the package ecosystem of the distro that people use. So if we can't help users keep tailscale up-to-date on Arch without breaking stuff, then we'll have to resort to docs and hope people read them.

Reply:
> https://wiki.archlinux.org/title/System_maintenance#Partial_upgrades_are_unsupported talks mostly about libraries. AFAIK the tailscale package itself doesn't contain any shared libraries and only depends on libc.
Is it still risky to update this kind of "leaf" package?

The command is not acting on the package at all.

`pacman -Sy tailscale` is the same as `pacman -Sy; pacman -S tailscale`. You are refreshing the databases and will pull inn updates relevant for the `tailscale` package. Currently `tailscale` depends on `glibc` so if you are so unlucky there is a new glibc version (which your system does not yet support), you are pulling inn both.

This will break your system.

> As for the general topic of auto-updates, is there any "standard" way to apply updates in the background for a rolling release like Arch? Other than writing a cronjob, of course.

You just don't do this at all on Arch. Arch is a rolling release distro where users needs to resolve the result of any updates. Stuffing this into a cronjob is just a disaster waiting to happen. Unattended upgrades has the same issue.

>We would really like to keep all tailscale clients on recent-ish versions, both for general bug/security fixes and for reducing the backwards-compatibility kludges in APIs that tailscale clients talk to.
But we also want to play nice with the package ecosystem of the distro that people use.

On Linux distros, how would implementing a completely new command encourage people to update more frequently? People are lazy when things work and updating unattended is bad behavior. Telling users to update more often is not really going to make them do it either.


> . So if we can't help users keep tailscale up-to-date on Arch without breaking stuff, then we'll have to resort to docs and hope people read them.

Arch assumes competent users, so this is completely fine to expect of them.

Reply:
To be clear, `pacman -Sy tailscale` is going to make all the next `pacman -S` operations dangerous until a full upgrade , `pacman -Syu`,  is completed.